### DISCUSSION TITLE
New IC request and feature idea

### DISCUSSION THREAD

**Discussion by akshaygs** [lcgamboa/picsimlab#71] [Q&A]
Hi,

Can you add 28 pin PIC16F1516 IC, PIC18F14Q10 
Also I want to use this IC to create a layout for a simple GPIO devboard like in the example you have provided.

But what I want to add new is every gpio element of the board
For example: LED's Anode pin or Buttons signal pin we can right click and chose to assign to a pin so as to keep it as flexible as possible for the users. I want to create this to help teach my students better.

<img width="304" alt="pcb" src="https://user-images.githubusercontent.com/33262039/221405773-78906d2f-7820-4fc9-be27-5c3582d97701.png">

If you can try adding the IC I will try and make the board as instructed in the website

Thank you for your time

**Comment by lcgamboa** [comment]
Hi @akshaygs ,

I will try to add these two microcontrollers as soon as I have time.  
Currently, there is support for 28-pin PICs: PIC16F18855, PIC16F1788, PIC16F886, PIC18F26K80, and PIC18F27K40. Does any of these not suit you?
I believe it is possible to implement the components so that they can be reconfigured.  It will work like the parts in the spare parts window .

**Reply by akshaygs** [reply]
Thank you very much for taking your time to add the ICs.

Sure, they would work. Since we already made the development boards and the learning content in the these ics, and also the price point of these ICs are desirable for learning.

As for the reconfigurable headers to all the gpio, the benefit here is the entire experience can be done on one board and one window and with the similar board that we provide during learning..

**Comment by lcgamboa** [comment]
@akshaygs,

I cannot find information about the PIC18F14Q10. Is the specification correct?

**Reply by akshaygs** [reply]
Sorry I meant PIC18F24Q10
https://www.microchip.com/en-us/product/PIC18F24Q10

**Comment by lcgamboa** [comment]
@akshaygs ,

support to PIC16F1516 and  PIC18F24Q10 added on commit https://github.com/lcgamboa/picsimlab/commit/c90dcc73c1ce5bd4c6917d5726da67e9b9d45006 .
